engine/src/flutter/lib/ui/geometry.dart

```diff
@@ -1743,6 +1743,261 @@ class RRect {
   }
 }
 
+/// An immutable rounded superellipse.
+///
+/// A rounded superellipse is a shape similar to a typical rounded rectangle
+/// ([RRect]), but with smoother transitions between the straight sides and the
+/// rounded corners. It resembles the `RoundedRectangle` shape in SwiftUI with
+/// the `.continuous` corner style. Technically, it is created by replacing the
+/// four corners of a superellipse (also known as a Lamé curve) with circular
+/// arcs.
+///
+/// The corner radius parameters used in this class corresponds to SwiftUI's
+/// `cornerRadius` parameter, which is close to, but not exactly equals to, the
+/// radius of the corner circles.
+class RSuperellipse {
+  /// Construct a rounded superellipse from its bounding box and a corner radius
```

Comment:
I'm going to come back and double-check the code in this new class. Moving along for now so I can get an initial review out.